### ISSUE TITLE
showing location of credentials

### ISSUE THREAD

**Issue by ahoward** [rubygems/rubygems#8373]
I am at a loss debugging and issue.... I was having auth issues (wrong key scope) and eventually worked around this.  However, I in the process I discovered that my .gem/credentials were not, in fact, stored in the location but, instead, where stored in `.local/share/gem/`

I cannot seem to find any reference to this online, or in the code so, I am befuddled.

1. Any thoughts on why `gem signin` would produce credentials there, instead of the normal spot and
2. Is there (there should be) a way to display which credentials location the `gem` command is using?

TY!

**Comment by duckinator** [comment]
Hi! This change happened a few years ago, when we started following the XDG standard. It should definitely be documented.

In practice, you can think of it as RubyGems and Bundler checking these 3 directories and going with the first they can use:

- if it exists: `$HOME/.gem/` (for backwards compatibility)
- if `$XDG_DATA_HOME` is defined: `$XDG_DATA_HOME/gem/` (to follow the XDG standard)
- in all other situations: `$HOME/.local/share/gem/` (also to follow the XDG standard)

(For anyone who's interested, the relevant code for this in [`Gem.user_dir`](https://github.com/rubygems/rubygems/blob/b2bf239/lib/rubygems/defaults.rb#L103-L109) and [`Gem.home`](https://github.com/rubygems/rubygems/blob/b2bf239147ba31ddd251bfac02534250d04a32ca/lib/rubygems/defaults.rb#L154-L156), both in `lib/rubygems/defaults.rb`.)

**Comment by ahoward** [comment]
Yes.  I figured this out.  Would definitely be nice to doc.  I could submit
a PR if helpful?

On a side note that standard, while useful, makes backups tricky since that
directory is FULL of craft, like logs, from other commands like gcloud.....
Easy enough to work around when docs are there but...

TY

**Comment by duckinator** [comment]
I opened #8375 about adding the information to `gem env`. I'm not sure where documentation for it would make sense to go, but I agree documenting it might be useful as well.

**Comment by hsbt** [comment]
@ahoward Please let me know https://github.com/ruby/rubygems/pull/8375 is not enough with your usecase.
